samtranslator/intrinsics/actions.py

```diff
@@ -333,11 +327,11 @@ def _handle_sub_action(self, input_dict, handler):  # type: ignore[no-untyped-de
         key = self.intrinsic_name
         sub_value = input_dict[key]
 
-        input_dict[key] = self._handle_sub_value(sub_value, handler)  # type: ignore[no-untyped-call]
+        input_dict[key] = self._handle_sub_value(sub_value, handler)
 
         return input_dict
 
-    def _handle_sub_value(self, sub_value, handler_method):  # type: ignore[no-untyped-def]
+    def _handle_sub_value(self, sub_value: Any, handler_method: Callable[[str, str], str]) -> Any:
```

Comment:
It seems to me that the function returns `Union[str, List[str]]`. Am I right? I think the purpose of this and similar functions in this PR is to transition from `Any` to 1 or 2 types

Reply:
How can you tell it's only `Union[str, List[str]]`? Isn't `Dict` also one of the viable type?